sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/validationstests/ValidationsTest.java

```diff
@@ -0,0 +1,140 @@
+        final String exceptionStr = testCase.getIsEnabled().getException();
+        return exceptionStr != null && !exceptionStr.isEmpty();
+    }
+
+    private boolean hasInput(ValidationTestCase testCase) {
+        final LinkedHashMap<String, Object> inputsMap = testCase.getInputs();
+        return inputsMap != null && !inputsMap.isEmpty();
+    }
+
+    private File[] getFileList(String folderPath, String fileNameFilter) {
+        final URL folderUrl = Thread.currentThread().getContextClassLoader().getResource(folderPath);
+        final File folderFile = new File(folderUrl.getFile());
+        return folderFile.listFiles(pathname -> pathname.getName().toLowerCase().contains(fileNameFilter));
+    }
+
+    private List<ValidationTestCase> readTestcasesFromFile(File testFile) throws IOException {
+        final String jsonString = Files.readString(testFile.toPath());
+        final CollectionType typeReference =
+            TypeFactory.defaultInstance().constructCollectionType(List.class, ValidationTestCase.class);
+        return OBJECT_MAPPER.readValue(jsonString, typeReference);
+    }
+
+    private LinkedHashMap<String, Object> readConfigurationFromFile(File sampleFile) throws IOException {
+        final String jsonString = Files.readString(sampleFile.toPath());
+        final LinkedHashMap<String, Object> configurations = OBJECT_MAPPER.readValue(jsonString, new TypeReference<>() {
+        });
+        final Object featureManagementSection = configurations.get("feature_management");
+        if (featureManagementSection.getClass().isAssignableFrom(LinkedHashMap.class)) {
+            return (LinkedHashMap<String, Object>) featureManagementSection;
+        }
+        return new LinkedHashMap<>();
+    }
+
+    private void runTestcases(File sampleFile, File testsFile) throws IOException {
+        // initialize feature manager
+        final FeatureManagementProperties managementProperties = new FeatureManagementProperties();
+        managementProperties.putAll(readConfigurationFromFile(sampleFile));
+        final FeatureManager featureManager = new FeatureManager(context, managementProperties, configProperties);
+
+        final List<ValidationTestCase> testCases = readTestcasesFromFile(testsFile);
+        for (ValidationTestCase testCase : testCases) {
+            if (hasException(testCase)) {   // TODO. Currently we didn't throw the exception when parameter is invalid
+                assertNull(managementProperties.getOnOff().get(testCase.getFeatureFlagName()));
+            } else {
+                if (hasInput(testCase)) { // Set inputs
+                    final Object userObj = testCase.getInputs().get(INPUTS_USER);
+                    final Object groupsObj = testCase.getInputs().get(INPUTS_GROUPS);
+                    final String user = userObj != null ? userObj.toString() : null;
+                    final List<String> groups = groupsObj != null ? (List<String>) groupsObj : null;
```

Comment:
Something seems to have changed recently in the build pipelines. Currently any warnings are making builds fail. Add a `@SuppressWarnings("unchecked")` to this for now.

Reply:
Updated